Update docs from feedbacks (#107)
unoplatform/uno.resizetizer: doc/using-uno-resizetizer.md

> [!TIP]
> To create an app, make sure to visit [our getting started tutorials](xref:Uno.GetStarted).

## How it works

> Resizeter uses a `svg` or `png` file as input. If a `svg` file is used, it will be re-scaled for different resolutions. The UnoImage, on iOS for example, will use the x2 and x3 corresponded sizes and add them on your project for you. If a `png` file is used, it will not be resized, but it will be added to your project and used as one size image. If you want to know all the scales that are used, you can check this [table](https://platform.uno/docs/articles/features/working-with-assets.html#scale).

Reviewer: ```suggestion
Resizetizer uses an `svg` or `png` file as input. If an `svg` file is used, it will be re-scaled for different resolutions. The `UnoImage`, on iOS for example, will use the x2 and x3 corresponded sizes and add them on your project for you. If a `png` file is used, it will not be resized, but it will be added to your project and used as one size image. If you want to know all the scales that are used, you can check this [table](https://platform.uno/docs/articles/features/working-with-assets.html#scale).
```

> Resizetizer uses an `svg` or `png` file as input. If an `svg` file is used, it will be re-scaled for different resolutions. The `UnoImage`, on iOS for example, will use the x2 and x3 corresponded sizes and add them on your project for you. If a `png` file is used, it will not be resized, but it will be added to your project and used as one size image. If you want to know all the scales that are used, you can check this [table](https://platform.uno/docs/articles/features/working-with-assets.html#scale).

Reviewer 2: ```suggestion
Resizetizer uses an `svg` or `png` file as input. If an `svg` file is used, it will be re-scaled for different resolutions. The `UnoImage`, on iOS, for example, will use the x2 and x3 corresponding sizes and add them to your project for you. If a `png` file is used, it will not be resized, but it will be added to your project and used as one size image. If you want to know all the scales that are used, you can check this [table](https://platform.uno/docs/articles/features/working-with-assets.html#scale).
```

> For `UnoIcon` and `UnoSplashScreen` the generated sizes will be the same as the ones used by the platform.

Reviewer: ```suggestion
For `UnoIcon` and `UnoSplashScreen`, the generated sizes will be the same as the ones used by the platform.
```

> With that we strongly encourage the usage of `svg` files, to take most of the benefits of the tool. And the rest of the docs you can assume that we are using `svg` files.

Reviewer: Can we specify what `that` is?

Reviewer 3: @pictos can you clarify what "that" is, so we can finalise these changes

> [!TIP]
> You can use the `Resize` property on UnoImage to force the resize of a `png` file. But be aware that the quality can be affected.

## Manual Installation

Uno.Resizeter is delivered [through NuGet](https://www.nuget.org/packages/Uno.Resizetizer). In order to install it you can either download it using your IDE (this will be shown in the next steps) or added directly on your `.csproj` as shown in the [NuGet page](https://www.nuget.org/packages/Uno.Resizetizer/).

> [!NOTE]
> If you're using the new template, you can skip this step because it is already included by default with Uno Platform 4.8 and later.
  

> ### 1. Installing Uno.Resizetizer

Reviewer 2: ```suggestion
### [Optional Step] Installing Uno.Resizetizer
```
@jeromelaban and @nickrandolph 
I think this step should be mentioned as optional instead of **Step 1** as most of the devs will use a template where `Uno.Resizetizer` is already installed. What do you think?

Reviewer 3: The Note is sufficient

Uno.Resizeter is delivered [through NuGet](https://www.nuget.org/packages/Uno.Resizetizer). In order to install it you can either download it using your IDE (this will be shown in the next steps) or added directly on your `.csproj` as shown in the [NuGet page](https://www.nuget.org/packages/Uno.Resizetizer/).

> > [!NOTE]

Reviewer 4: In my opinion, still find it confusion for the user to keep this as step 1 with just a note above it. Can we just make this an optional step instead of step 1?

Author: cc: @nickrandolph

Reviewer 3: I'm not sure what this is referring to?

The next sections will show how to use it for each use case.

> [!WARNING]
> All the assets used by Uno.Resizetizer should be lower case and don't have special characters. You can use `_` to separate words.

> > This is because the assets are used on different platforms and some of them have limitations on the characters that can be used.

Author: For now, we can mention it on docs, there's already an error message explaining that. If we decide to change something it will be in another PR

</ItemGroup>
```

You can also make specific files to be `UnoImage` using VisualStudio, by right-clicking on the file and selecting `Properties` and then `Build Action` and selecting `UnoImage`. The image below shows how it looks like:

![UnoImage Build Action](Assets/UnoImage_BuildAction.png)

> 

Reviewer 4: * In the App Class library, create a folder called `Assets` (if doesn't exist) and then create a folder called `Images`. **We now need to add assets to this folder.**

Please specify that we need to add `svg` file as it is not clear for multiple devs while going through the documentation that they should had svg files and not png during this step.

Author: @agneszitte-nventive We mentioned on the docs at the first paragraph that users should provide `svg` images, I can't see value repeating the same information over and over again

Author: cc: @nickrandolph

Reviewer 3: In general, we're currently not explaining what resizetizer actually does, only instructions for how to use it. A couple of suggestions:
- In the first paragraph of this doc we should separate the sentence talking about svg into its own paragraph. Make a note that Resizetizer will work with svg and png (others??) but that it will only resize svgs into appropriate scaled png files.
- Provide a more general explanation of what Resizetizer does - it takes a single image and generates the necessary image files for each platform. 
- For UnoImage, state that this should be added to the class library (or shared library for older projects)
- For UnoIcon and UnoSplashscreen, this should be added to base.props (or shared library for older projects) so that it's added to the target projects. 

At beginning of these instructions, reiterate that unless specified otherwise, all images referred to are svg files.

### 3. Using the assets on the project

* `UnoImage` assets can now be used just like any regular image. For example:

```xml
<Image Width="300"
       Height="300"

>        Source="Assets\\Images\\my_image.png" />

Author: The `ms-appx` schema will not work.
cc: @nickrandolph

>        Source="Assets\\Images\\my_image.png" />

Author: @nickrandolph with that, which one should we present on docs? The `ms-appx` one or just the folder?

>        Source="Assets\\Images\\my_image.png" />

Reviewer 3: Relative path - which is what's in the template

`UnoIcon` is the build action for the app icon. There should only be one per application. The `UnoIcon` accepts two assets, one that represents the `Foreground` and another that represents the `Background`. During the generation phase, those files will be merged into one `.png` image.

> During the creation of your `svg` file, please remember to make the `ViewBox` bigger than the `Foreground` and `Background` images, not adding an extra space could make the app icon doesn't look good on some platforms. We recommend to add a 30% extra space on each side. This will be enough for Resizetizer to work with padding and margins.

Reviewer 4: @jeromelaban , @pictos
 Not sure if this is enough information for the dev to understand what to do with the svg file if they are not used to it (as in this case it would imply editing the data of the svg and adjusting the ViewBox properties. Note also that not all svg file contains a ViewBox in some cases)

Author: @agneszitte-nventive as I mentioned during the stand up today, when I finish my spike around the `svg` I'll update the docs with a small guide on how they should create the `svg`

`UnoIcon` is the build action for the app icon. There should only be one per application. The `UnoIcon` accepts two assets, one that represents the `Foreground` and another that represents the `Background`. During the generation phase, those files will be merged into one `.png` image.

During the creation of your `svg` file, please remember to make the `ViewBox` bigger than the `Foreground` and `Background` images, not adding an extra space could make the app icon doesn't look good on some platforms. We recommend to add a 30% extra space on each side. This will be enough for Resizetizer to work with padding and margins.

> ### 4. Configuring the project to use generated app icon

Reviewer 2: @pictos and @nickrandolph 
I think we should mention in this section the `ForegroundScale` property (and `AndroidForegroundScale`, `iOSForegroundScale`,...) needed in case the icon needs some adjustments on some platforms

* `BaseSize`: It's the size that will be used to perform the scaling of the image. The default value is the size of the asset. So if you feel that your SplashScreen doesn't look right you can tweak this value.

> * `Color`: It's the background color that will be used to fill the empty space on the final SplashScreen asset. The default value is `#FFFFFF`(white).

Reviewer: ```suggestion
* `Color`: It's the background color that will be used to fill the empty space on the final SplashScreen asset. The default value is `#FFFFFF` (white).
```

         Color="#512BD4" />
```

> We recommend to add the `UnoSplashScreen` on `base.props` because this file is imported by all head projects, that way, you don't need to add the same configuration on each head project.

Reviewer 4: ```suggestion
We recommend adding the `UnoSplashScreen` on `base.props` because this file is imported by all head projects, that way, you don't need to add the same configuration on each head project.
```

    <item name="android:windowSplashScreenBackground">@color/uno_splash_color</item>
    <item name="android:windowBackground">@drawable/uno_splash_image</item>
    <item name="android:windowSplashScreenAnimatedIcon">@drawable/uno_splash_image</item>

    <!-- Image at the footer -->
    <!-- This is not required in order to make the Splash screen work -->

>     <!-- For more info please see: https://developer.android.com/develop/ui/views/launch/splash-screen#set-theme -->

Reviewer 2: I think this section for the Image footer for the Android Splashscreen should be in another subheader section for Android-specific details only. What do you think @jeromelaban and @nickrandolph?

Reviewer: I'm not seeing a reason to move it somewhere else, this seems related to the current android block.

Reviewer 3: I think it's ok where it is
